### PR TITLE
Update doc of batch size requirements for DP

### LINKED ISSUE #5039: torch.nn.DataParallel supporting unequal sizes
As documented [here](http://pytorch.org/docs/master/_modules/torch/nn/parallel/data_parallel.html):
```
The batch size should be larger than the number of GPUs used. It should
    also be an integer multiple of the number of GPUs so that each chunk is the
    same size (so that each GPU processes the same number of samples).
```
To use `torch.nn.DataParallel`, people should carefully set the batch size according to the number of gpus they plan to use, otherwise it will pop up errors. 

This issue becomes more subtle when using `torch.utils.data.DataLoader` with `drop_last=False` by default. As the total number of training/validation samples varies with the dataset, the size of the last batch of data loaded by `torch.utils.data.DataLoader` is easy to become indivisible by the number of GPUs (e.g., 2,3,4,8,...).

A feature request would be:
supporting `torch.nn.DataParallel` with batch size indivisible by the number of GPUs used.

### PR DESCRIPTION
**Pull request by Stonesjtu**
Fix #5039

### CODE REVIEW

**Review by apaszke — COMMENTED** [review]
Why would that affect performance? I can’t see a connection

**Comment by Stonesjtu** [comment]
@apaszke For we need to synchronize all the gradients among GPU cards for one batch, GPU with small ` batch-size` would wait for the GPUs with large `batch-size` to complete.

e.g. There are 4 GPUs available.
1. `batch-size=80`, each card actually computes `80/4 = 20` samples for one batch.
2. `batch-size=81`, we have card 0,1,2 with `batch-size=81/4 = 21`. and card 3 with `batch-size = 18`. In this case, increasing batch-size to `84` reduces *the number of batches* (total samples / batch-size) while costing the same computing time during each batch.

**Comment by SsnL** [comment]
@Stonesjtu It's probably true in the ideal case, but realistically, many things can matter, e.g. different GPU models, other GPU loads, some data sample needs more computation, etc.

**Comment by Stonesjtu** [comment]
I can't agree more, predicting the workload can get even more complicated since pytorch ultilizes dynamic graph which is built on every data batch. But  `batch_size` is at least the most general metric of workload that can be easily obtained before actual **forward** and **backward** .

**Comment by SsnL** [comment]
Yes, that's right. What I was trying to say that divisible `batch_size` doesn't necessarily have impact on performance in practice, so to not confuse the users, probably excluding that part will be better.

**Review by ssnl — APPROVED** [review]
LGTM. Thanks @Stonesjtu !

**Comment by songtaoshi** [comment]
Hi, since it has low impact, would it be possible to make the imbalanced distribution ???
